qualtran/bloqs/factoring/ecc/ec_add_test.py

```diff
@@ -12,13 +12,456 @@
+        lam=step_2['lam'],
+    )
+    ret2 = bloq.decompose_bloq().call_classically(
+        ctrl=step_2['ctrl'],
+        a=step_2['a'],
+        b=step_2['b'],
+        x=step_2['x'],
+        y=step_2['y'],
+        lam=step_2['lam'],
+    )
+    assert ret1 == ret2
+
+    step_3 = _ECAddStepThree(n=n, mod=p, window_size=m).on_classical_vals(
+        ctrl=step_2['ctrl'],
+        a=step_2['a'],
+        b=step_2['b'],
+        x=step_2['x'],
+        y=step_2['y'],
+        lam=step_2['lam'],
+    )
+    bloq = _ECAddStepFour(n=n, mod=p, window_size=m)
+    ret1 = bloq.call_classically(x=step_3['x'], y=step_3['y'], lam=step_3['lam'])
+    ret2 = bloq.decompose_bloq().call_classically(x=step_3['x'], y=step_3['y'], lam=step_3['lam'])
+    assert ret1 == ret2
+
+    step_4 = _ECAddStepFour(n=n, mod=p, window_size=m).on_classical_vals(
+        x=step_3['x'], y=step_3['y'], lam=step_3['lam']
+    )
+    bloq = _ECAddStepFive(n=n, mod=p, window_size=m)
+    ret1 = bloq.call_classically(
+        ctrl=step_3['ctrl'],
+        a=step_3['a'],
+        b=step_3['b'],
+        x=step_4['x'],
+        y=step_4['y'],
+        lam=step_4['lam'],
+    )
+    ret2 = bloq.decompose_bloq().call_classically(
+        ctrl=step_3['ctrl'],
+        a=step_3['a'],
+        b=step_3['b'],
+        x=step_4['x'],
+        y=step_4['y'],
+        lam=step_4['lam'],
+    )
+    assert ret1 == ret2
+
+    step_5 = _ECAddStepFive(n=n, mod=p, window_size=m).on_classical_vals(
+        ctrl=step_3['ctrl'],
+        a=step_3['a'],
+        b=step_3['b'],
+        x=step_4['x'],
+        y=step_4['y'],
+        lam=step_4['lam'],
+    )
+    bloq = _ECAddStepSix(n=n, mod=p)
+    ret1 = bloq.call_classically(
+        f1=step_2['f1'],
+        f2=step_1['f2'],
+        f3=step_1['f3'],
+        f4=step_1['f4'],
+        ctrl=step_5['ctrl'],
+        a=step_5['a'],
+        b=step_5['b'],
+        x=step_5['x'],
+        y=step_5['y'],
+    )
+    ret2 = bloq.decompose_bloq().call_classically(
+        f1=step_2['f1'],
+        f2=step_1['f2'],
+        f3=step_1['f3'],
+        f4=step_1['f4'],
+        ctrl=step_5['ctrl'],
+        a=step_5['a'],
+        b=step_5['b'],
+        x=step_5['x'],
+        y=step_5['y'],
+    )
+    assert ret1 == ret2
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize(
+    ['n', 'm'], [(n, m) for n in range(7, 10) for m in range(1, n + 1) if n % m == 0]
+)
+@pytest.mark.parametrize(
+    'a,b',
+    [
+        (15, 13),
+        (2, 10),
+        (8, 3),
+        (12, 1),
+        (6, 6),
+        (5, 8),
+        (10, 15),
+        (1, 12),
+        (3, 0),
+        (1, 5),
+        (10, 2),
+        (0, 0),
+    ],
+)
+@pytest.mark.parametrize(
+    'x,y',
+    [
+        (15, 13),
+        (2, 10),
+        (8, 3),
+        (12, 1),
+        (6, 6),
+        (5, 8),
+        (10, 15),
+        (1, 12),
+        (3, 0),
+        (1, 5),
+        (10, 2),
+        (0, 0),
+    ],
+)
+def test_ec_add_steps_classical(n, m, a, b, x, y):
```

Comment:
the parameterization here seems a little gratuitous. do we need the outer product of all possible n x m? What are we probing with different n values? it either fits or it doesn't, no? how about picking some extremal pairs? Do we need the complete outer product of point1 x point2? Maybe cross all possible point1 with some select extremal point2's

Reply:
I pretty much halved the testing by removing a big chunk of the pairs. I did it in a way such that all 4 bugs from https://github.com/quantumlib/Qualtran/issues/1461 would still trigger if I made the frees non-dirty. Let me know if you want me to further reduce the tests. All-in-all it runs in 1 min 30 s on my pc with the slow tests running.